### PR TITLE
add erode otsu labeling function

### PR DESCRIPTION
**Pull request by StRigaud**
closes https://github.com/clEsperanto/pyclesperanto/issues/136

### CODE REVIEW

**Review by StRigaud — COMMENTED** [review]
voided

**Review by haesleinhuepf — CHANGES_REQUESTED** [review]
Hey @StRigaud ,

this is looking great! Just one thing: the name should be `eroded_otsu_labeling`, 
see: https://github.com/clEsperanto/pyclesperanto_prototype/blob/413e3fadbff7d5db07c98d614cd5e2d2a94ae38c/pyclesperanto_prototype/_tier9/_eroded_otsu_labeling.py#L4

So you're missing a `d` (and the test has the correct name) ;-)

File: clic/src/tier7.cpp [hunk 1]
```diff
@@ -13,17 +13,28 @@
...
-//                       bool                    resize) -> Array::Pointer
-// {
-//   auto transform_matrix = AffineTransform(transform);
-//   return apply_affine_transform(src, dst, transform_matrix, interpolate, resize);
-// }
+auto
+eroded_otsu_labeling_func(const Device::Pointer & device,
+                          const Array::Pointer &  src,
+                          Array::Pointer          dst,
+                          int                     number_of_erosion,
```
Comment (haesleinhuepf) [inline clic/src/tier7.cpp:20]:
```suggestion
                          int                     number_of_erosions,
```
One more little thing that makes it API-compatible with the `_prototype`

File: clic/src/tier7.cpp [hunk 2]
```diff
@@ -13,17 +13,28 @@
...
+{
+  tier0::create_like(src, dst, dType::UINT32);
+  auto           blurred = tier1::gaussian_blur_func(device, src, nullptr, outline_sigma, outline_sigma, outline_sigma);
+  auto           binary = tier4::threshold_otsu_func(device, blurred, nullptr);
+  Array::Pointer eroded1 = nullptr;
+  Array::Pointer eroded2 = nullptr;
+  tier0::create_like(binary, eroded1);
+  tier0::create_like(binary, eroded2);
+  binary->copy(eroded1);
+  for (int i = 0; i < number_of_erosion; i++)
```
Comment (haesleinhuepf) [inline clic/src/tier7.cpp:31]:
```suggestion
  for (int i = 0; i < number_of_erosions; i++)
```

**Review by haesleinhuepf — CHANGES_REQUESTED** [review]
and one more tiny name change please

File: clic/include/tier7.hpp [hunk 3]
```diff
@@ -10,6 +10,41 @@
...
+ * @return Array::Pointer
+ *
+ * @link [0] https://github.com/biovoxxel/bv3dbox (BV_LabelSplitter.java#L83)
+ * @link [1] https://zenodo.org/badge/latestdoi/434949702
+ */
+auto
+eroded_otsu_labeling_func(const Device::Pointer & device,
+                          const Array::Pointer &  src,
+                          Array::Pointer          dst,
+                          int                     number_of_erosion,
```
Comment (haesleinhuepf) [inline clic/include/tier7.hpp:44]:
```suggestion
                          int                     number_of_erosions,
```

**Review by haesleinhuepf — APPROVED** [review]
LGTM

**Pull request merged**